GVRf/Framework/src/org/gearvrf/GVRMesh.java

```diff
@@ -66,8 +67,7 @@ protected final boolean registerWrapper() {
      *            Array containing the packed vertex data.
      */
     public void setVertices(float[] vertices) {
-        isNonNull(vertices);
-        isValidData(vertices.length, 3);
+        validOtherCheck(INPUT_TYPE_VERTICES, vertices, vertices.length, 3);
```

Comment:
> Actually I am also quite sure about this part. Because there are general two types of check operations. One is for all kinds of vector, the other one is for the data other than vector.

Sorry, you lost me. I'm not sure what you're referring to ...


>  Preconditions class hold all those Strings ... is easy to manage and extend

Actually, it's not easy to manage or extend. If everyone puts their strings in a shared class, you can end up with merge conflicts and/or naming chaos. Every change invalidates (re-compiles) every module that uses it. Keeping the tag values in the client means that a new tag can be local to the client module, and adding a new tag only invalidates one module.

> let the class itself to hold those constant String ... may cause some duplicates. 

Java does a fine job of making each and every reference to a constant `"string"` refer to the same run-time `String` instance. The names would be local.

> Jon, How do you think?

I've told you twice, now ...